elements/lisk-p2p/src/peer/base.ts

```diff
@@ -510,7 +512,9 @@ export class Peer extends EventEmitter {
 	public async fetchPeers(): Promise<ReadonlyArray<P2PPeerInfo>> {
 		try {
 			const response: P2PResponsePacket = await this.request({
-				procedure: REMOTE_RPC_GET_PEERS_LIST,
+				procedure: isLowerVersionPeer(this._peerInfo as P2PDiscoveredPeerInfo)
+					? REMOTE_RPC_GET_PEERS_LIST
+					: REMOTE_RPC_GET_MINIMAL_PEERS_LIST,
 			});
 
 			return validateBasicPeersInfoList(response.data);
```

Comment:
Maybe we should always remove other properties (except for ip/ipAddress and wsPort), even if the peer is the old version and it gives us the full list. The idea for the original feature was that we can't trust PeerInfo which we haven't connected to so that is why we don't store the other fields. Maybe we should keep this aspect.